python/dgl/graphbolt/impl/neighbor_sampler.py

```diff
@@ -222,11 +328,21 @@ def _compact_per_layer(self, minibatch):
 class FetcherAndSampler(MiniBatchTransformer):
     """Overlapped graph sampling operation replacement."""
 
-    def __init__(self, sampler, stream, executor, buffer_size):
+    def __init__(
+        self,
+        sampler,
+        num_cached_edges,
+        threshold,
+        stream,
+        executor,
+        buffer_size,
+    ):
         datapipe = sampler.datapipe.fetch_insubgraph_data(
-            sampler, stream, executor
+            sampler, num_cached_edges, threshold, stream, executor
         )
         datapipe = datapipe.buffer(buffer_size).wait_future().wait()
+        if num_cached_edges > 0:
```

Comment:
What will happen if num_cached_edges > 1 while using CPU sampling?

Reply:
CPU sampling should not enter this code path at all. This code path is reserved for GPU sampling.

Reply:
This code path is enabled only when overlap_graph_fetch = True in gb.Dataloader.